pkg/api/secretgenerator/secretgenerator_test.go

```diff
@@ -0,0 +1,142 @@
+package secretgenerator
+
+import (
+	"errors"
+	"fmt"
+	"path/filepath"
+	"sort"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+)
+
+func TestLoadConfigFromPath(t *testing.T) {
+	testcases := []struct {
+		name          string
+		config        string
+		expected      Config
+		expectedError error
+	}{
+		{
+			name:          "file not exist",
+			expectedError: errors.New("open testdata/TestLoadConfigFromPath/file_not_exist.yaml: no such file or directory"),
+		},
+		{
+			name: "no_parameters",
+			expected: Config{
+				{
+					ItemName:    "Item1",
+					Attachments: []FieldGenerator{{Name: "Attachment1", Cmd: "echo -n Attachment1"}},
+				},
+			},
+		},
+		{
+			name: "single parameter",
+			expected: Config{
+				{
+					ItemName:    "Item1",
+					Fields:      []FieldGenerator{{Name: "Field1", Cmd: "echo -n Field1"}},
+					Attachments: []FieldGenerator{{Name: "Attachment1", Cmd: "echo -n Attachment1"}},
+					Password:    "echo -n password1", Notes: "Note1",
+					Params: map[string][]string{
+						"FieldNum": {"1"},
+					},
+				},
+			},
+		},
+		{
+			name: "single parameter with multiple values",
+			expected: Config{
```

Comment:
Well so ideally both the in and out of the test is a file, so we have a file `config_some_test.yaml` and `config_some_test_expanded.yaml` so you can just look at the files and also autogenerate them via `pkg/testhelper.CompareWithFixture`